gm_dotnet_managed/GmodNET/GmodNetModuleAssemblyLoadContext.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Runtime.Loader;
+using System.IO;
+using System.Reflection;
+using System.Linq;
+using GmodNET.API;
+
+namespace GmodNET
+{
+    internal class GmodNetModuleAssemblyLoadContext : ModuleAssemblyLoadContext
+    {
+        private AssemblyDependencyResolver resolver;
+        private string module_name;
+        private Func<ModuleAssemblyLoadContext, string, IntPtr> customNativeLibraryResolver;
+
+        public override string ModuleName
+        { 
+            get
+            {
+                return module_name;
+            }
+        }
+
+        public override Func<ModuleAssemblyLoadContext, string, IntPtr> CustomNativeLibraryResolver
+        {
+            get
+            {
+                return customNativeLibraryResolver;
+            }
+        }
+
+        public override void SetCustomNativeLibraryResolver(Func<ModuleAssemblyLoadContext, string, IntPtr> resolver)
+        {
+            customNativeLibraryResolver = resolver;
+        }
+        
+        internal GmodNetModuleAssemblyLoadContext(string module_name) : base(isCollectible: true)
+        {
+            this.module_name = module_name;
+            resolver = new AssemblyDependencyResolver("garrysmod/lua/bin/Modules/"+module_name+"/"+module_name+".dll");
+            customNativeLibraryResolver = null;
+        }
+
+        protected override System.Reflection.Assembly Load(System.Reflection.AssemblyName assemblyName)
+        {
+            if(assemblyName.Name == "GmodNET.API")
+            {
+                return null;
+            }
+
+            string path = resolver.ResolveAssemblyToPath(assemblyName);
+            if (path == null || path == string.Empty)
+            { 
+                return null;
```

Comment:
Would it be better to throw an exception here?

Reply:
@AaronJThompson  This method returns `null` in order to give CoreCLR resolve this dependency using default process as described here: https://docs.microsoft.com/en-us/dotnet/api/system.runtime.loader.assemblyloadcontext.load?view=netcore-3.1#remarks

Reply:
Ah okay, I wasn't aware of this! I'll have a re-read through and check anywhere else for nulls/IntPtr.Zero's as I know they can sometimes cause unexpected behaviour